gesos-image-build.pipeline

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env groovy
+
+@Library(['PPCmanifest','security-ci-commons-shared-lib']) _
+
+def repoStatusReportUrl
+def gitCommitStatusContext = 'GE SOS Image Build'
+
+pipeline {
+    agent {
+        docker {
+            image 'openjdk:8'
+            label 'dind'
+        }
+    }
+    parameters {
+        string(
+            name: 'IAM_CONTAINER_CONFIG_BRANCH_NAME',
+            defaultValue: BRANCH_NAME,
+            description: 'iam-container-config repo branch name to use for GE SOS image build'
+        )
+        string(
+            name: 'IAM_CONTAINER_CONFIG_FOLDER_NAME',
+            defaultValue: 'uaa',
+            description: 'Name of container config folder with Dockerfile and prebuild script in the ' +
+                         'iam-container-config repo to use for GE SOS image build'
+        )
+        string(
+            name: 'IMAGE_TAG',
+            defaultValue: 'BUILD_TAG_FROM_BRANCH_NAME',
+            description: 'Tag for docker image to push to ECR'
+        )
+        string(
+            name: 'IMAGE_NAME',
+            defaultValue: 'iam-uaa',
+            description: 'Name for docker image to push to ECR'
+        )
+    }
+    environment {
+        GESOS_API_KEY = credentials('GESOS_API_KEY')
+        GITHUB_CREDENTIALS = credentials('GITHUB_CREDENTIALS')
+    }
+    stages {
+        stage('Github status configuration') {
+            steps {
+                script {
+                    commitSha = sh(script: "git rev-parse HEAD", returnStdout: true).trim()
+                    repoStatusReportUrl = sh(script: 'git config --get remote.origin.url', returnStdout: true).trim()
+                                                                                                              .replaceFirst('^https://([^/]+)/([^.]+)(\\.git)?$',
+                                                                                                                            'https://$1/api/v3/repos/$2/statuses/' + commitSha)
+                }
+            }
+        }
+        stage('Trigger GE SOS build pipeline') {
+            steps {
+                script {
+                    setBuildStatus(repoStatusReportUrl, gitCommitStatusContext, 'in progress', 'pending')
```

Comment:
As agreed during team discussion, explicit reporting is needed so each checkpoint status, e.g, branch build, GE SOS build, etc, is recorded in GitHub.  Otherwise, GitHub reports the status of the last checkpoint.